esp-hal/src/uart.rs

```diff
@@ -1030,10 +1030,8 @@ where
     pub fn new_typed(
         uart: impl Peripheral<P = T> + 'd,
         config: Config,
-        rx: impl Peripheral<P = impl PeripheralInput> + 'd,
-        tx: impl Peripheral<P = impl PeripheralOutput> + 'd,
     ) -> Result<Self, ConfigError> {
-        UartBuilder::new(uart).with_tx(tx).with_rx(rx).init(config)
```

Comment:
As far as I remember, the tx-rx order here matters. You should try and swap the pins in the failing tests, if we can't figure out why this is...

Reply:
Hmm, for some reason I can't reproduce the CI error, I tried different pins and it still passes for me 🤔